adafruit_bno055.py

```diff
@@ -401,12 +403,27 @@ def _write_register(self, register, data):  # pylint: disable=arguments-differ
 
     def _read_register(self, register, length=1):  # pylint: disable=arguments-differ
         self._uart.write(bytes([0xAA, 0x01, register, length]))
-        time.sleep(0.1)
+        now = time.time()
+        while self._uart.in_waiting < length + 2 and time.time() - now < 0.25:
+            pass
         resp = self._uart.read(self._uart.in_waiting)
+
+        if resp[0] != 0xBB:  # Recursion
```

Comment:
This is not recursion because recursion would call _read_register. Why did you add this? It wasn't needed before.

Reply:
My initial plan was to actually do that, but I couldn't think of a good way to add in an exit condition so it didn't just keep going until it reached the recursion limit, and I guess I forgot to remove the comment.

It's needed because the, when running in UART mode, even in the version that's currently on master, it raises a UART read error every 3 or 4 reads when you're trying to use the simpletest. I think that before it may have mostly been tested by getting a single value, such as acceleration, at a time. That being said, even though I know the majority of people aren't using this with UART, I am quite surprised that I haven't seen anyone else encountering that issue, so it may be worthwhile for some other people with the sensor to do some testing to ensure that issue doesn't only affect me.

Reply:
Do the errors go away if you increase the timeout? If you really need it, I think you could just have an outer loop and count how many tries you want to do before raising an error. No need to copy things a second time.